pageserver/src/object_store.rs

```diff
@@ -14,6 +14,27 @@ pub struct ObjectKey {
     pub buf_tag: BufferTag,
 }
 
+/// A single version of an Object.
+pub struct ObjectVersion {
+    pub lsn: Lsn,
+    pub value: Box<[u8]>,
+}
+
+/// An iterator over `ObjectVersion`
+pub type ObjectsIter<'a> = dyn Iterator<Item = ObjectVersion> + 'a;
```

Comment:
maybe `ObjectVersions`/`ObjectVersionsIter` to match with the function name

Reply:
OK with me.

Reply:
I started making this change and realized `ObjectVersions` seems strange to me because one instance is only one version. I could go either way on the iterator name.

Reply:
To clarify, I think the `ObjectVersion` struct name above makes sense. Mostly thinking about the iterator name switching to `ObjectVersions`/`ObjectVersionIter`. If we decide to include `Iter` for the changes below, we should probably follow that pattern here too

Reply:
And then I remembered something else: `object_repository.rs` already has a related struct called `ObjectVersionIter`. It's a bit confusing to have two structs with the same or similar name. Not sure which should change.

Reply:
Ah I see. There's also an `ObjectTimeline` struct in `object_repository.rs` which makes the `TimelineObject` naming I mentioned below more confusing